ksql-parser/src/main/java/io/confluent/ksql/parser/rewrite/StatementRewriteForRowtime.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.parser.rewrite;
+
+import io.confluent.ksql.parser.tree.BetweenPredicate;
+import io.confluent.ksql.parser.tree.ComparisonExpression;
+import io.confluent.ksql.parser.tree.DereferenceExpression;
+import io.confluent.ksql.parser.tree.Expression;
+import io.confluent.ksql.parser.tree.FunctionCall;
+import io.confluent.ksql.parser.tree.Node;
+import io.confluent.ksql.parser.tree.QualifiedName;
+import io.confluent.ksql.parser.tree.StringLiteral;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class StatementRewriteForRowtime {
+  private final Expression expression;
+
+  public StatementRewriteForRowtime(final Expression expression) {
+    this.expression = Objects.requireNonNull(expression, "expression");
+  }
+
+  public static boolean requiresRewrite(final Expression expression) {
+    return expression.toString().contains("ROWTIME");
+  }
+
+  public Expression rewriteForRowtime() {
+    return (Expression) new RewriteWithTimestampTransform().process(expression, null);
+  }
+
+  private static class TimestampRewriter extends StatementRewriter {
+    private static final String DATE_PATTERN = "yyyy-MM-dd";
+    private static final String TIME_PATTERN = "HH:mm:ss.SSS";
+    private static final String PATTERN = DATE_PATTERN + "'T'" + TIME_PATTERN;
+
+    @Override
+    public Expression visitFunctionCall(final FunctionCall node, final Object context) {
+      return (Expression) new StatementRewriter().process(node, context);
```

Comment:
Does this line do anything other than return a duplicate of the function call node? I think `StatementRewriter` should be `abstract` as its meant as a base class for rewrites. In itself it doesn't actually change the query, right?

Reply:
It stops the rewriter from further processing anything within a function. So, if the expression looks like `ROWTIME > FOO('2017-01-01 ' + 100)  + '2019-01-01'`, then ` '2019-01-01'` will get wrapped but `'2017-01-01'` doesn't.